### PR TITLE
feat(compass-indexes): Poll rolling indexes where available, add status column COMPASS-8217  COMPASS-8329

### PR DESCRIPTION
**Pull request by lerouxb**
Screenshot 2024-10-03 at 13 39 08

### CODE REVIEW

**Review by gribnoysup — APPROVED** [review]
Looks neat! Tried running this with my changes, works pretty good (ran into some quirks with how in progress indexes behave for rolling indexes now that we can create them from UI, but it's not really related to changes here in any way)

File: packages/compass-indexes/src/components/regular-indexes-table/regular-indexes-table.tsx [hunk 1]
```diff
@@ -192,23 +230,58 @@ type MappedRegularIndex = RegularIndex & { compassIndexType: 'regular-index' };
...
-  const mappedIndexes: MappedRegularIndex[] = indexes.map((index) => {
-    return { ...index, compassIndexType: 'regular-index' };
-  });
+  const rollingIndexNames = rollingIndexes.reduce<Record<string, true>>(
+    (result, index) => {
+      result[index.indexName] = true;
+      return result;
+    },
+    {}
+  );
```
[the remark this answers is not in the document]
Comment (lerouxb) [reply packages/compass-indexes/src/components/regular-indexes-table/regular-indexes-table.tsx:254]: I always forget about set for some reason.

File: packages/compass-indexes/src/components/regular-indexes-table/status-field.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        <Badge data-testid="index-ready" variant={BadgeVariant.Green}>
+          Ready
+        </Badge>
+      )}
+
+      {status === 'building' && (
+        <Badge data-testid="index-building" variant={BadgeVariant.Blue}>
+          Building
+        </Badge>
+      )}
```
[the remark this answers is not in the document]
Comment (lerouxb) [reply packages/compass-indexes/src/components/regular-indexes-table/status-field.tsx:65]: Makes sense. I can add a tooltip.

[the remark this answers is not in the document]
Comment (lerouxb) [reply packages/compass-indexes/src/components/regular-indexes-table/status-field.tsx]: Done. See the screenshot in the description.

**Pull request merged**